src/compiler/binder.ts

```diff
@@ -327,33 +329,37 @@ namespace ts {
                 }
 
                 if (symbol.flags & excludes) {
-                    if (node.name) {
-                        node.name.parent = node;
+                    if (symbol.isDiscardable) {

```

Comment:
nit: you may want to mention this change in the comment [above ](https://github.com/Microsoft/TypeScript/pull/10123/files#diff-7455dc48a52c4974a3e3671fcfb1dfb9R308)